packages/orchestration/src/exos/remote-chain-facade.js

```diff
@@ -148,8 +148,9 @@ const prepareRemoteChainFacadeKit = (
               connectionInfo,
             } = this.state;
             if (!icqEnabled) {
-              throw Fail`Queries not available for chain ${chainId}`;
+              throw Fail`Queries not available for chain ${q(chainId)}`;
             }
+            mustMatch(msgs, M.arrayOf(ICQMsgShape), 'ICQMsgShape');
```

Comment:
this should be validated by the guard. I see you changed `chainFacadeMethods` to be less strict. please override in the local guard to get the more strict version. e.g.,

```js
      public: M.interface('RemoteChainFacade', {
        ...chainFacadeMethods,
          query: M.call(M.arrayOf(M.ICQMsgShape())).returns(VowShape),
      }),
```